utils.go

```diff
@@ -72,17 +72,17 @@ func GetRootlessRuntimeDir(rootlessUid int) (string, error) {
 	runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
 	if runtimeDir == "" {
 		tmpDir := fmt.Sprintf("/run/user/%d", rootlessUid)
-		st, err := os.Stat(tmpDir)
-		if err == nil && int(st.Sys().(*syscall.Stat_t).Uid) == os.Getuid() && st.Mode().Perm() == 0700 {
+		st, err := system.Stat(tmpDir)
+		if err == nil && int(st.UID()) == os.Getuid() && st.Mode() == 0700 {
 			return tmpDir, nil
 		}
 	}
 	tmpDir := fmt.Sprintf("%s/%d", os.TempDir(), rootlessUid)
 	if err := os.MkdirAll(tmpDir, 0700); err != nil {
 		logrus.Errorf("failed to create %s: %v", tmpDir, err)
 	} else {
-		st, err := os.Stat(tmpDir)
-		if err == nil && int(st.Sys().(*syscall.Stat_t).Uid) == os.Getuid() && st.Mode().Perm() == 0700 {
+		st, err := system.Stat(tmpDir)
+		if err == nil && int(st.UID()) == os.Getuid() && st.Mode() == 0700 {
```

Comment:
This should compile, which is good, but `st.UID()` is hardwired to always return 0, while `os.Getuid()` is documented to always return -1, so our attempt to create this directory ends up either failing and producing an error, or succeeding and then being ignored.

Reply:
My preference is to avoid creating a directory that we're just going to ignore.

Reply:
Why would stating of the directory return UID==0?

Reply:
`pkg/system`'s `Stat()` returns a pointer to a `StatT`, which on Windows systems only ever returns 0 from its `UID()` method.

Reply:
Do you want to make this OS Specific then?

Reply:
we end up using this code only when `os.Geteuid() != 0`, on Windows it returns always `-1`.  Could we just check `os.Geteuid()` value and treat `-1` differently?